packages/expo-notifications/src/NotificationsEmitter.ts

```diff
@@ -121,3 +126,30 @@ export async function getLastNotificationResponseAsync(): Promise<NotificationRe
   const mappedResponse = response ? mapNotificationResponse(response) : response;
   return mappedResponse;
 }
+
+/**
+ * Clears the notification response that was received most recently. May be used in cases
+ * where the app selects a route based on the notification response, and it is undesirable
+ * to continue to select the route after the response has already been handled.
+ *
+ * If a component is using the [`useLastNotificationResponse`](#useLastNotificationResponse) hook,
+ * this call will also clear the value returned by the hook.
+ *
+ * @return A promise that resolves if the native call was succesful.
+ */
+export async function clearLastNotificationResponseAsync(): Promise<void> {
+  if (!NotificationsEmitterModule.clearLastNotificationResponseAsync) {
+    throw new UnavailabilityError('ExpoNotifications', 'getLastNotificationResponseAsync');
+  }
+  emitter.emit(didClearNotificationResponseEventName, []);
+  return await NotificationsEmitterModule.clearLastNotificationResponseAsync();
+}
+
+/**
+ * @hidden
+ */
+export function addNotificationResponseClearedListener(listener: () => void): EventSubscription {
+  return emitter.addListener<void>(didClearNotificationResponseEventName, () => {
+    listener();
+  });
```

Comment:
```suggestion
  return emitter.addListener<void>(didClearNotificationResponseEventName, listener);
```

Not a big difference in this case (saving the function allocation), but I find this equally readable.